test/e2e/serial/workload_placement_test.go

```diff
@@ -151,3 +165,94 @@ func filterNodesWithMoreThanOneNUMAZone(nodes []corev1.Node, nodeResourceTopolog
 
 	return ret, nil
 }
+
+func getWorkloadRequirements(nodeResourceTopologies []nrtv1alpha1.NodeResourceTopology, nodes []corev1.Node) (corev1.ResourceList, error) {
+	for _, nrt := range nodeResourceTopologies {
+		// for all the zones
+		for _, zone := range nrt.Zones {
+			// find zones bigger than the current one
+			zonesMap, err := zonesBiggerThan(zone, nodeResourceTopologies)
+			if err != nil {
+				return nil, err
+			}
+			if len(zonesMap) == 0 {
+				// if no other zone matches but the zone itself.
+				// find if any node but the one this zone comes from can match the requests
+				for _, node := range nodes {
+					if node.Name == nrt.Name {
+						continue
+					}
+					//convert zone resources into a resource request
+					zoneResources := resourceInfoListToResourceList(&zone.Resources)
+					if nodeAllocatableResourcesMatchesRequest(node, zoneResources) {
+						// if there is another node then this is our request
+						// because any pod with this resource request could be sheduled
+						// both to the node or to the NUMA zone, so it is ok for our testing purposes
+						return zoneResources, nil
+					}
+				}
+			}
+		}
+	}
+	return nil, fmt.Errorf("not suitable requirement found")
+}
+
+func resourceInfoListToResourceList(origin *nrtv1alpha1.ResourceInfoList) corev1.ResourceList {
+	ret := make(corev1.ResourceList)
+	for _, resInfo := range *origin {
+		ret[corev1.ResourceName(resInfo.Name)] = resInfo.Available
+	}
+	return ret
+}
+
+func nodeAllocatableResourcesMatchesRequest(node corev1.Node, request corev1.ResourceList) bool {
+
+	for reqName, reqQuantity := range request {
+		nodeQuantity, found := node.Status.Allocatable[reqName]
+		if !found {
+			return false
+		}
+		if nodeQuantity.Cmp(reqQuantity) < 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func zonesBiggerThan(zone nrtv1alpha1.Zone, nrts []nrtv1alpha1.NodeResourceTopology) (map[string][]nrtv1alpha1.Zone, error) {
+
+	ret := make(map[string][]nrtv1alpha1.Zone)
+	for _, nrt := range nrts {
+		for _, z := range nrt.Zones {
+			cmp, err := zoneCmp(zone, z)
+			if err != nil {
+				return nil, err
+			}
+			if cmp < 0 {
+				ret[nrt.Name] = append(ret[nrt.Name], z)
+			}
+		}
+	}
+	return ret, nil
+}
+
+func zoneCmp(z1, z2 nrtv1alpha1.Zone) (int, error) {
+	equal := true
+	for _, res := range z1.Resources {
+		qty, found := e2enrt.FindResourceAvailableByName(z2.Resources, res.Name)
+		if !found {
+			return 0, fmt.Errorf("resource not found: %q", res.Name)
+		}
+		cmp := res.Available.Cmp(qty)
+		if cmp < 0 {
+			// once one resource is bigger we considerate the zone as bigger
```

Comment:
this is a fragile assumption though. Do we need this granularity? I'd try to review the overall flow in such a way we don't need to compare zones. It's fine (and required!) to compare the pod footprint (resources required) with zones, but we should be wary to compare zones, as we can fall in edge cases.

Example: two equal NUMA zones. 32 cpus and 32Gi memory each. We reserve cpus for system tasks from zone 0. We happen to have devices on zone 0 as well. So, which zone is "bigger" and why?
Point in case: there is not an universally right decision.

Reply:
Yeah ... I get your point and I do not like this code either ... comparing zones as units it is difficult as they have very different and not related resources.
The point here is that Pod required resources are the result we want, we need to create a pod with such a requirements that it could fit into a single numa zone in say NodeA but also into would fit into NodeB but using resources from more than one NUMA zone, and check the pod is properly scheduled.

I was trying to know the "biggest" NUMA zone to assign the pod the zone resources as required resources to ensure the only way to schedule the pod is using the NUMA zone or a complete node  ... 
but maybe it is enough to assign to the pod any resources to make it fit on any NUMA zone and then check the resources on one numa zone have decreased in that same amount.